src/executables/csql_support.c

```diff
@@ -239,17 +239,97 @@ csql_invoke_system (const char *command)
     }
 }
 
+
+/*
+ * csql_invoke_formatter()
+ *   return: CSQL_SUCCESS/CSQL_FAILURE
+ *
+ * Note:
+ *   copy command editor buffer into temporary file and
+ *   invoke formatter. After the format is finished,
+ *   read the file into editor buffer.
+ */
+int
+csql_invoke_formatter ()
+{
+  auto[before_filename, before_fileptr] = filesys::open_temp_file ("bef_fmt_");
+  if (before_fileptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+  filesys::auto_delete_file before_file_del (before_filename.c_str ());
+  filesys::auto_close_file before_file (before_fileptr);
+
+  if (csql_edit_write_file (before_file.get ()) == CSQL_FAILURE)
+    {
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
+    }
+
+  auto[after_filename, after_fileptr] = filesys::open_temp_file ("aft_fmt_");
+  if (after_fileptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+  filesys::auto_delete_file after_file_del (after_filename.c_str ());
+  filesys::auto_close_file after_file (after_fileptr);
+
+  char *cmd = csql_get_tmp_buf (strlen (csql_Formatter_cmd) + 1 + before_filename.size () + 1 + after_filename.size ());
+  if (cmd == NULL)
+    {
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
+    }
+  fclose (after_file.release ());
+  sprintf (cmd, "%s %s > %s", csql_Formatter_cmd, before_filename.c_str (), after_filename.c_str ());
+
+  before_file.reset (fopen (before_filename.c_str (), "r"));
+  if (!before_file)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+
+  after_file.reset (fopen (after_filename.c_str (), "r"));
+  if (!after_file)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+
+  if (system (cmd) != 0)
+    {
+      csql_Error_code = CSQL_ERR_FORMAT;
```

Comment:
free_and_init (cmd) is also required when an error occurs in system().

Reply:
That part is missing. I'll correct it.
Thank you.